### DISCUSSION TITLE
Swap fee on amountOut

### DISCUSSION THREAD

**Discussion by rahulthakkar796** [Elnaril/uniswap-universal-router-decoder#24] [Q&A] [answered]
Could anyone please provide an example of how to swap custom ERC20 tokens for ETH, or vice versa, while applying a 1% swap fee on the `amountOut` and ensuring that the fee is transferred to a specific address for each swap?

**Comment by Elnaril** [comment]
Hello @rahulthakkar796 
This is not done through the Universal Router, but through custom smart contracts or the token contracts.
Closing the discussion as not related to the UUR SDK.

**Reply by rahulthakkar796** [reply]
Hello, thank you for your response. Actually, it's possible using the Universal Router. Take a look at this transaction: https://etherscan.io/tx/0x43eb84da467de5cb1a3f3017e1b491cf0d56070ba0069726b453147e3f833d43
 It essentially transfers a 1% swap fee from `amountOut` to a specific address. You can try out any swap on https://www.pond0x.com/. They perform the same process with all pairs, and they use the Universal Router to achieve this. I'm currently trying to understand how it's done using the Universal Router, as there doesn't seem to be any documentation available on this.

**Comment by Elnaril** [comment]
Ah ok, I see: you're talking of the `PAY_PORTION` and `SWEEP` functions. 👍
These are commands that you add to the transaction you send to the UR, but now I understand you call it "fees", since in your case, it's done by the DEX.
=> I'm re-opening the discussion.

Unfortunately, these functions are not yet supported by this SDK ...
If you are interested in the SDK supporting them, please feel free to open a [feature request](https://github.com/Elnaril/uniswap-universal-router-decoder/issues/new/choose).

**Comment by Elnaril** [comment, marked as answer]
Hello @rahulthakkar796 

v1.0.0 has been released and it adds support for the Universal Router functions `PAY_PORTION` and `SWEEP`.

So here is an example of how to swap ETH for an erc20 token (here UNI) with a 1% fee sent to another address:

```python
# Buying for 1 eth of uni from v2 and send 1% of out_amount to account_2
v2_path = [weth_address, uni_address]
v2_in_amount = 1 * 10 ** 18  # 1 eth
out_amount = 381140129360952845496
v2_out_amount = int(out_amount * 0.995)  # 0.5% slippage
fees = 100  # 1%
encoded_input = (
    codec
    .encode
    .chain()
    .wrap_eth(FunctionRecipient.ROUTER, v2_in_amount)
    .v2_swap_exact_in(FunctionRecipient.ROUTER, v2_in_amount, v2_out_amount, v2_path, payer_is_sender=False)
    .pay_portion(FunctionRecipient.CUSTOM, uni_address, fees, account_2.address)
    .sweep(FunctionRecipient.SENDER, uni_address, 0)
    .build(codec.get_default_deadline())
)
```

Please, let me know if it answers your question.

**Reply by Elnaril** [reply]
@rahulthakkar796 , does it answer your question?
